### PR TITLE
wip: add support for container capabilities monitoring 

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-ebpf

> /area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:
This PR tries to add support for container capabilities monitoring for Docker and CRI compatible container runtimes. A new field `container.capabilities` prints out a space-separated list of capabilities granted to the container. `container.capability` can be used to understand if a specific capability was granted.

**Which issue(s) this PR fixes**:

Fixes [#1812](https://github.com/falcosecurity/falco/issues/1812)

**Special notes for your reviewer**:
The same result could be achieved by parsing the file `/proc/<pid>/status`. However, that is avoidable since we are already retrieving information from the sockets used to communicate with the container runtime. Data about capabilities are there in each response and ready to be used. Moreover, since we are not making any new requests to the container runtime's socket, I expect no performance impact.

**Does this PR introduce a user-facing change?**:

```release-note
update: add support for container capabilities monitoring introducing the container.capabilities and container.capability fields
```

### CODE REVIEW

File: userspace/libsinsp/container_engine/docker/async_source.cpp [hunk 1]
```diff
@@ -795,3 +867,82 @@ bool docker_async_source::parse_docker(const docker_lookup_request& request, sin
...
+		added_capabilities.push_back(c);
+	}
+	for(const auto& cap : drop)
+	{
+		string c = cap.asString();
+		if(parse_capability(c))
+		{
+			all_dropped = true;
+			break;
+		};
```
Comment (FedeDP) [inline userspace/libsinsp/container_engine/docker/async_source.cpp:895]: `;`...typo? Same goes for the above one!

**Review by FedeDP — APPROVED** [review]
This is a great PR, thanks!

File: userspace/libsinsp/container_engine/docker/async_source.cpp (hunk 1 above)
Comment (loresuso) [reply userspace/libsinsp/container_engine/docker/async_source.cpp]: Fixed now, thanks!

**Comment by loresuso** [comment]
Also added the possibility to extract a single capability, e.g `container.capability[CAP_SYS_ADMIN]`. 
This might come in handy when writing Falco rules, for instance. 

/retest

**Comment by loresuso** [comment]
Closing in favor of https://github.com/falcosecurity/libs/pull/234

**Pull request closed without merging**